article-analyzer/src/main/java/co/inventorsoft/academy/homework/service/impl/CategorySaverServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package co.inventorsoft.academy.homework.service.impl;
+
+import co.inventorsoft.academy.homework.service.CategorySaverService;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.lang.reflect.Type;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+@Component
+@AllArgsConstructor
+public class CategorySaverServiceImpl implements CategorySaverService {
+    private static final String CATEGORIES_FILE_PATH = "article-analyzer/src/main/resources/categories.json";
```

Comment:
it is a good fit for moving the constant into application.properties, the same for articles and users

Reply:
done